src/Http/Routing/src/EndpointRoutingMiddleware.cs

```diff
@@ -106,6 +112,13 @@ private Task SetRoutingAndContinue(HttpContext httpContext)
             }
             else
             {
+                // Raise an event if the route matched
+                if (_diagnosticListener.IsEnabled() && _diagnosticListener.IsEnabled(DiagnosticsRouteMatchedKey))
+                {
+                    // We're just going to send the HttpContext since it has all of the relevant information
+                    _diagnosticListener.Write(DiagnosticsRouteMatchedKey, httpContext);
```

Comment:
1. Is it correct that this event will fire for both: WebApi and MVC?
2. How do we get route from HttpContext?
3. AppInsights uses https://github.com/aspnet/Mvc/blob/master/src/Microsoft.AspNetCore.Mvc.Core/MvcCoreDiagnosticSourceExtensions.cs events to know about routing today. Should we stop listening to those?

Reply:
1. It will fire for any framework that matches using endpoint routing: MVC, razor pages, gRPC, SignalR
2. Yes
3. It should still be used for legacy apps. You should be able to listen for both side-by-side. I can't imagine why an ASP.NET Core would use both.